indexed_gzip/zran.c

```diff
@@ -1113,10 +1098,11 @@ int _zran_init_zlib_inflate(zran_index_t *index,
          */
         if (point->bits > 0) {
 
-            ret = getc(index->fd);
+            ret = getc_(index->fd, index->f);
 
-            if (ret == -1)
+            if (ret == -1 && ferror_(index->fd, index->f)) {
```

Comment:
This appears to have been a bug in the original implementation of `zran.c` -- it's not enough to check that `getc` returns `-1`, because `getc` can return `-1` if it returns the character `255` (`\xFF`). To check if `getc` returned an error, you need to check that `getc` returns `-1` and that `ferror` is set.

This problem was causing my tests to fail with `no_fds` set to `True`, so the tests do cover it.